### ISSUE TITLE
Issue with sklearn

### ISSUE THREAD

**Issue by nickschw2** [gpp-rnd/rs3#9]
I am receiving the following error from running this basic code upon installing rs3

```
from rs3.seq import predict_seq
context_seqs = ['GACGAAAGCGACAACGCGTTCATCCGGGCA', 'AGAAAACACTAGCATCCCCACCCGCGGACT']
predict_seq(context_seqs, sequence_tracr='Hsu2013')
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/rs3/seq.py", line 53, in predict_seq
    seq_predictions = model.predict(featurized_sgrnas)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/lightgbm/sklearn.py", line 899, in predict
    predict_params = self._process_params(stage="predict")
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/lightgbm/sklearn.py", line 674, in _process_params
    if self._n_classes > 2:
       ^^^^^^^^^^^^^^^^^^^
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

I believe this might be associated with the fact that sklearn is now deprecated? I'm not completely sure, but any help would be greatly appreciated.

**Comment by ericmalekos** [comment]
@nickschw2 I was having the same problem. It seems to stem from sklearn.py in lightgbm 4.3.0. I solved the problem by downgrading:

`pip install lightgbm==3.3.5`

**Comment by PeterDeWeirdt** [comment]
I fixed the max version of lightgbm to 3.3.5 in the newest release so the model should work out of the box now. Thanks for pointing out this dependency issue @nickschw2 and thanks for the easy fix @ericmalekos
